core/processor/src/main/java/io/quarkus/annotation/processor/generate_doc/ConfigDocItemScanner.java

```diff
@@ -163,18 +163,55 @@ public void addConfigRoot(final PackageElement pkg, TypeElement clazz) {
             }
 
             final String fileName = computeExtensionDocFileName(member);
-            final List<ConfigDocItem> previousExtensionConfigDocKeys = foundExtensionConfigurationItems.get(fileName);
+            final List<ConfigDocItem> existingConfigDocItems = foundExtensionConfigurationItems.get(fileName);
 
-            if (previousExtensionConfigDocKeys == null) {
+            if (existingConfigDocItems == null) {
                 foundExtensionConfigurationItems.put(fileName, configDocItems);
             } else {
-                previousExtensionConfigDocKeys.addAll(configDocItems);
+                // merge two config doc items list
+                mergeConfigItemsIntoExistingOnes(existingConfigDocItems, configDocItems);
             }
         }
 
         return foundExtensionConfigurationItems;
     }
 
+    private void mergeConfigItemsIntoExistingOnes(List<ConfigDocItem> existingConfigDocKeys,
+            List<ConfigDocItem> configDocItems) {
+        for (ConfigDocItem configDocItem : configDocItems) {
+            if (configDocItem.isConfigKey()) {
+                existingConfigDocKeys.add(configDocItem);
+            } else {
+                boolean configSectionAdded = addSectionIntoPreviousExistingConfigItems(configDocItem.getConfigDocSection(),
+                        existingConfigDocKeys);
+                if (!configSectionAdded) {
+                    existingConfigDocKeys.add(configDocItem);
+                }
+            }
+        }
+    }
+
+    /**
+     * returns true if section added into one of the existing config properties, false otherwise
+     */
+    private boolean addSectionIntoPreviousExistingConfigItems(ConfigDocSection section, List<ConfigDocItem> configDocItems) {
+        for (ConfigDocItem configDocItem : configDocItems) {
+            if (configDocItem.isConfigKey()) {
+                continue;
+            }
+
+            ConfigDocSection configDocSection = configDocItem.getConfigDocSection();
+            if (configDocSection.equals(section)) {
+                configDocSection.addConfigDocItems(section.getConfigDocItems());
+                return true;
+            } else if (addSectionIntoPreviousExistingConfigItems(section, configDocSection.getConfigDocItems())) {
```

Comment:
I'm not sure to fully understand this one?

Reply:
Ah this should have been updated. I forget to force push the change. I’ll do it later on.